Kernel/Heap/kmalloc.cpp

```diff
@@ -243,6 +298,33 @@ void* kmalloc_eternal(size_t size)
     return ptr;
 }
 
+void* kmalloc_eternal(size_t size)
+{
+    kmalloc_verify_nospinlock_held();
+
+    size = round_up_to_power_of_two(size, sizeof(void*));
+
+    ScopedSpinLock lock(s_lock);
+    // Note: Only Eternal heaps should be in the static eternal heap of the kernel,
+    // therefore each pointer must of an EternalHeap and only it!
+    EternalHeap* eternal_heaps_metadata = reinterpret_cast<EternalHeap*>(kmalloc_eternal_heaps);
+    if (auto* ptr = eternal_heaps_metadata[g_dynamic_eternal_heaps_count].allocate(size); ptr != nullptr)
+        return ptr;
+
+    // Note: It's very unlikely that the MemoryManager was no initialized at this stage!
```

Comment:
```suggestion
    // Note: It's very unlikely that the MemoryManager was not initialized at this stage!
```

very unlikely or impossible? :)

Reply:
Well, in theory it's possible, but extremely unlikely to happen. If it happens, we did something really wrong.